domain/model/watcher_test.go

```diff
@@ -0,0 +1,211 @@
+
+	// Add a cloud so we can set the model cloud.
+	cloudSt := cloudstate.NewState(dbTxnRunnerFactory)
+	err = cloudSt.CreateCloud(context.Background(), userName, uuid.MustNewUUID().String(),
+		cloud.Cloud{
+			Name:      "my-cloud",
+			Type:      "ec2",
+			AuthTypes: cloud.AuthTypes{cloud.AccessKeyAuthType, cloud.UserPassAuthType},
+			Regions: []cloud.Region{
+				{
+					Name: "my-region",
+				},
+			},
+		})
+
+	c.Assert(err, jc.ErrorIsNil)
+
+	// Add a cloud credential so we can set the model cloud credential.
+	cred := credential.CloudCredentialInfo{
+		Label:    "label",
+		AuthType: string(cloud.AccessKeyAuthType),
+		Attributes: map[string]string{
+			"attr1": "foo",
+			"attr2": "bar",
+		},
+	}
+	credSt := credentialstate.NewState(dbTxnRunnerFactory)
+	_, err = credSt.UpsertCloudCredential(
+		context.Background(), corecredential.Key{
+			Cloud: "my-cloud",
+			Owner: userName,
+			Name:  "my-cloud-credential",
+		},
+		cred,
+	)
+	c.Assert(err, jc.ErrorIsNil)
+
+	err = bootstrap.CreateDefaultBackends(coremodel.IAAS)(context.Background(), dbTxnRunner, dbTxnRunner)
+	c.Assert(err, jc.ErrorIsNil)
+}
+
+func (s *watcherSuite) SetUpTest(c *gc.C) {
+	s.ControllerSuite.SetUpTest(c)
+	s.userUUID = usertesting.GenUserUUID(c)
+	s.userName = usertesting.GenNewName(c, "test-user")
+	insertModelDependencies(c, s.TxnRunnerFactory(), s.TxnRunner(), s.userUUID, s.userName)
+}
+
+func (s *watcherSuite) TestWatchControllerDBModels(c *gc.C) {
+	ctx := context.Background()
+	logger := loggertesting.WrapCheckLog(c)
+	watchableDBFactory := changestream.NewWatchableDBFactoryForNamespace(s.GetWatchableDB, "model")
+	watcherFactory := domain.NewWatcherFactory(watchableDBFactory, logger)
+	st := state.NewState(func() (database.TxnRunner, error) { return watchableDBFactory() })
+
+	modelService := service.NewWatchableService(st, nil, loggertesting.WrapCheckLog(c), watcherFactory)
+
+	// Create a controller service watcher.
+	watcher, err := modelService.WatchActivatedModels(ctx)
+	c.Assert(err, jc.ErrorIsNil)
+
+	modelName := "test-model"
+	var modelUUID coremodel.UUID
+	var modelUUIDStr string
+	var activateModel func(ctx context.Context) error
+
+	harness := watchertest.NewHarness(s, watchertest.NewWatcherC(c, watcher))
+
+	// Verifies that watchers do not receive any changes when newly unactivated models are created.
+	harness.AddTest(func(c *gc.C) {
+		// Create a new unactivated model named test-model.
+		modelUUID, activateModel, err = modelService.CreateModel(ctx, domainmodel.GlobalModelCreationArgs{
+			Cloud:       "my-cloud",
+			CloudRegion: "my-region",
+			Credential: corecredential.Key{
+				Cloud: "my-cloud",
+				Owner: s.userName,
+				Name:  "my-cloud-credential",
+			},
+			Name:          modelName,
+			Owner:         s.userUUID,
+			SecretBackend: juju.BackendName,
+		})
+		c.Assert(err, jc.ErrorIsNil)
+		modelUUIDStr = modelUUID.String()
+
+		// Ensure that the initial activated status of model is indeed false.
+		var testModel dbModel
```

Comment:
I don't think we need these last line of this function. We already have test coverage for this and repeating this here confuses the point of the test.

Reply:
I see, sure will remove this thanks tom!